packages/discovery-react-components/src/utils/document/transformEnrichment.ts

```diff
@@ -145,23 +143,18 @@ function getAllAttributesInRelation({ relations }: Relations): Attributes[] {
 
 // Function checks if attribute is valid and has location data in it.
 // Since have observed noisy data before so this function is required.
-function isValidAttribute(attr: Attributes): boolean {
-  if (attr && attr.location && attr.location.begin && attr.location.end) {
-    return true;
-  }
-  return false;
+function hasLocationData(attr: Attributes): boolean {
+  return !!(attr && attr.location && attr.location.begin && attr.location.end);
```

Comment:
Should be able to do optional chaining now:

`return !!(attr?.location?.begin && attr?.location?.end)`

However, that still won't always work, because `begin` or `end` (well, probably only `begin`) could be zero, which would be seen as **falsy**. Probably better to check that they are not `undefined` using `typeof`.

Reply:
We can now use **Nullish Coalescing Operator - ??** which will take care of it.

Reply:
Well, `??` is more of an addition to `||`, but you're using `&&` here.

Reply:
Hmm.. for some reason VSCode is giving errors to ES2020 features. Have you configured anything for that?

Reply:
Oh, I thought we supported it. Perhaps ask in Slack.

Reply:
When TS, prettier upgrade PR goes in, will add the optional chaining in the same PR.

Reply:
This is still wrong: see my comment above after "However". `begin` can be zero, in which case this will incorrectly return `false`.